packages/desktop-client/src/components/reports/ReportSidebar.js

```diff
@@ -0,0 +1,411 @@
+  categories,
+  selectedCategories,
+  setSelectedCategories,
+}) {
+  function onChangeMode(cond) {
+    setMode(cond);
+    if (cond === 'time') {
+      if (graphType === 'TableGraph') {
+        setTypeDisabled([0]);
+      } else {
+        setTypeDisabled([3]);
+        if (['Net'].includes(balanceType)) {
+          setBalanceType('Expense');
+        }
+      }
+      if (graphType === 'BarGraph') {
+        setGraphType('StackedBarGraph');
+      }
+      if (['AreaGraph', 'DonutGraph'].includes(graphType)) {
+        setGraphType('TableGraph');
+        //setViewLegend(false);
+      }
+      if (['Month', 'Year'].includes(groupBy)) {
+        setGroupBy('Category');
+      }
+    } else {
+      if (graphType === 'StackedBarGraph') {
+        setGraphType('BarGraph');
+      } else {
+        setTypeDisabled([]);
+      }
+    }
+  }
+
+  function onChangeSplit(cond) {
+    setGroupBy(cond);
+    if (mode === 'total') {
+      if (graphType !== 'TableGraph') {
+        setTypeDisabled(!['Month', 'Year'].includes(groupBy) ? [] : ['Net']);
+      }
+    }
+    if (['Net'].includes(balanceType) && graphType !== 'TableGraph') {
+      setBalanceType('Expense');
+    }
+  }
+
+  return (
+    <View
+      style={{
+        width: 200,
+        paddingTop: 10,
+        paddingRight: 10,
+        flexShrink: 0,
+      }}
+    >
+      <View style={{ flexShrink: 0 }}>
+        <View
+          style={{
+            flexDirection: 'row',
+            marginBottom: 5,
+            alignItems: 'center',
+          }}
+        >
+          <Text>
+            <strong>Display</strong>
+          </Text>
+        </View>
+        <View
+          style={{
+            flexDirection: 'row',
+            padding: 5,
+            alignItems: 'center',
+          }}
+        >
+          <Text style={{ width: 40, textAlign: 'right', marginRight: 5 }}>
+            Mode:
+          </Text>
+          <ModeButton
+            selected={mode === 'total'}
+            onSelect={() => onChangeMode('total')}
+          >
+            Total
+          </ModeButton>
+          <ModeButton
+            selected={mode === 'time'}
+            onSelect={() => onChangeMode('time')}
+          >
+            Time
+          </ModeButton>
+        </View>
+        <View
+          style={{
+            flexDirection: 'row',
+            padding: 5,
+            alignItems: 'center',
+          }}
+        >
+          <Text style={{ width: 40, textAlign: 'right', marginRight: 5 }}>
+            Split:
+          </Text>
+          <Select
+            value={groupBy}
+            onChange={e => onChangeSplit(e)}
+            options={ReportOptions.groupBy.map(option => [
+              option.description,
+              option.description,
+            ])}
+            disabledKeys={
+              mode === 'time'
+                ? [5, 6]
+                : graphType === 'AreaGraph'
+                ? [1, 2, 3, 4, 6]
+                : [6]
+            }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏these are quite cryptic as well.

Reply:
Those needed translation. Thanks.

I'll be creating a "disabledList" in the next PR. This should organize and put into one place rather than the mess it is right now.